pkg/prometheus/agent/operator.go

```diff
@@ -1376,3 +1381,10 @@ func (c *Operator) checkAgentAuthorization(ctx context.Context) (bool, error) {
 
 	return true, nil
 }
+
+func (c *Operator) checkCRDInstallation(ctx context.Context) error {
+	crdClient := c.kclient.(*kubernetes.Clientset).RESTClient().Get().AbsPath("/apis/apiextensions.k8s.io/v1/customresourcedefinitions")
```

Comment:
We can use `k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset`.

https://github.com/prometheus-operator/prometheus-operator/blob/fa361d45b63b424495ac933769d484fbf9e9b7cb/test/framework/framework.go#L91-L94

Reply:
hmmm, feels a bit weird to instantiate another client when kclient.

Do you think the method below is more readable?

```go
func (c *Operator) checkCRDInstallation(ctx context.Context) error {
	cfg, err := k8sutil.NewClusterConfig(c.config.Host, c.config.TLSInsecure, &c.config.TLSConfig)
	if err != nil {
		return errors.Wrap(err, "instantiating cluster config failed")
	}

	client, err := kubernetes.NewForConfig(cfg)
	if err != nil {
		return errors.Wrap(err, "instantiating kubernetes client failed")
	}

	crdClient := client.RESTClient().Get().AbsPath("/apis/apiextensions.k8s.io/v1/customresourcedefinitions")
	res := crdClient.Name("prometheusagents.monitoring.coreos.com").Do(ctx)

	return res.Error()
}
```